Assets/ShooterProject/Scripts/Weapons/Weapon.cs

```diff
@@ -0,0 +1,152 @@
+using ShooterProject.Scripts.General;
+using System.Collections;
+using UnityEngine;
+using UnityEngine.InputSystem;
+using UnityEngine.XR.Interaction.Toolkit;
+
+namespace ShooterProject.Scripts.Weapons
+{
+
+	public class Weapon : MonoBehaviour
+	{
+		#region Fields
+
+		[SerializeField]
+		private WeaponParts _weaponParts;
+
+		[SerializeField]
+		private WeaponParams _weaponParams;
+
+		[SerializeField]
+		private WeaponShootingEffects _weaponShootingEffects;
+
+		private bool _isSelected = false;
+		private bool _coolDownOver = true;
+		private Coroutine _workingShootingCoroutine;
+		private GameObjectsPool _impactsPool;
+
+		#endregion
+
+		#region Properties
+		public bool IsSelected => _isSelected;
+		private bool _canShoot => _isSelected;
+		#endregion
```

Comment:
Добавь пустые строки
```suggestion
		#region Properties
		
		public bool IsSelected => _isSelected;
		private bool _canShoot => _isSelected;
		
		#endregion
```